lib/fluent/plugin/out_bigquery.rb

```diff
@@ -2,374 +2,303 @@
-
-    def initialize
-      super
-      require 'multi_json'
-      require 'google/apis/bigquery_v2'
-      require 'googleauth'
-      require 'active_support/json'
-      require 'active_support/core_ext/hash'
-      require 'active_support/core_ext/object/json'
-
-      # MEMO: signet-0.6.1 depend on Farady.default_connection
-      Faraday.default_connection.options.timeout = 60
-    end
-
-    def configure(conf)
-      if conf["method"] == "load"
-        configure_for_load(conf)
-      else
-        configure_for_insert(conf)
+        buffer_config = conf.elements("buffer")[0]
+        return unless buffer_config
+        buffer_config["@type"]                       = "file"        unless buffer_config["@type"]
+        buffer_config["flush_mode"]                  = :interval     unless buffer_config["flush_mode"]
+        buffer_config["chunk_limit_size"]            = 1 * 1024 ** 3 unless buffer_config["chunk_limit_size"] # 1GB
+        buffer_config["queue_length_limit"]          = 32            unless buffer_config["queue_length_limit"]
       end
-      super
-
-      case @method
-      when :insert
-        extend(InsertImplementation)
-      when :load
-        raise Fluent::ConfigError, "'template_suffix' is for only `insert` mode, instead use 'fetch_schema_table' and formatted table name" if @template_suffix
-        extend(LoadImplementation)
-      else
-        raise Fluent::ConfigError "'method' must be 'insert' or 'load'"
+
+      # Available methods are:
+      # * private_key -- Use service account credential from pkcs12 private key file
+      # * compute_engine -- Use access token available in instances of ComputeEngine
+      # * json_key -- Use service account credential from JSON key
+      # * application_default -- Use application default credential
+      config_param :auth_method, :enum, list: [:private_key, :compute_engine, :json_key, :application_default], default: :private_key
+
+      ### Service Account credential
+      config_param :email, :string, default: nil
+      config_param :private_key_path, :string, default: nil
+      config_param :private_key_passphrase, :string, default: 'notasecret', secret: true
+      config_param :json_key, default: nil, secret: true
+
+      # see as simple reference
+      #   https://github.com/abronte/BigQuery/blob/master/lib/bigquery.rb
+      config_param :project, :string
+
+      # dataset_name
+      #   The name can be up to 1,024 characters long, and consist of A-Z, a-z, 0-9, and the underscore,
+      #   but it cannot start with a number or underscore, or have spaces.
+      config_param :dataset, :string
+
+      # table_id
+      #   In Table ID, enter a name for your new table. Naming rules are the same as for your dataset.
+      config_param :table, :string, default: nil
+      config_param :tables, :string, default: nil # TODO: use :array with value_type: :string
```

Comment:
`:list` & `value_type: :string` is better for such case. It handles comma-separated string list, or JSON array string appropriately.